incubator/hnc/e2e/tests/issues_test.go

```diff
@@ -112,4 +117,40 @@ var _ = Describe("Issues", func() {
 		runShouldContain("CritParentMissing: missing parent", 1, "kubectl hns tree", nsChild)
 		runShouldContain("CritAncestor", 1, "kubectl hns describe", nsSubChild)
 	})
+
+	It("Should clear CannotUpdate conditions in descendants when a hierarchy changes - issue #605", func() {
+		// Setting up hierarchy with rolebinding that HNC doesn't have permission to copy
+		mustRun("kubectl create ns", nsParent)
+		mustRun("kubectl create ns", nsChild)
+		mustRun("kubectl create ns", nsSubChild)
+		mustRun("kubectl create ns", nsSubSubChild)
+		mustRun("kubectl hns set", nsChild, "--parent", nsParent)
+		mustRun("kubectl hns set", nsSubChild, "--parent", nsChild)
+		mustRun("kubectl hns set", nsSubSubChild, "--parent", nsSubChild)
+		// cluster-admin is the highest-powered ClusterRole and HNC is missing some of its permissions, so it cannot propagate it.
+		mustRun("kubectl create rolebinding cluster-admin-rb -n", nsParent, "--clusterrole='cluster-admin' --serviceaccount="+nsParent+":default")
+		// We put 30s sleep here because - In issue #605, our current object controller won't reconcile the objects in
+		// descendant of a namespace who changes its parent. However, in reality we found that the objects
+		// (greatgrandchild/cluster-admin-rb in this test) eventually get reconciled, triggered by controller-runtime.
+		// If sleeping shorter here, controller-runtime will enqueue the object faster since it has an exponential backoff.
+		// With sleeping 30s, we should see the object gets reconciled after around 8s, triggered by controller-runtime.
+		// While after fixing #605, we should see it gets reconciled and the obsolete conditions are gone immediately.
+		// See discussion - https://github.com/kubernetes-sigs/multi-tenancy/pull/617#issuecomment-610588109
+		time.Sleep(30 * time.Second)
+		// Tree should show CannotPropagateObject in parent and CannotUpdateObject in child and subchild
+		runShouldContain("CannotPropagateObject", 1, "kubectl hns describe", nsParent)
+		runShouldContain("CannotUpdateObject", 1, "kubectl hns describe", nsChild)
+		runShouldContain("CannotUpdateObject", 1, "kubectl hns describe", nsSubChild)
+		// Remove the grandchild to avoid removing CannotPropagate condition in parent or CannotUpdate condition in child.
+		// Verify that the conditions in grandchild and greatgrandchild are gone.
+		mustRun("kubectl hns set", nsSubChild, "--root")
+		// We use 10 because it takes around 9s here before the controller-runtime retry from an exponential backoff
+		// after the 30s sleep earlier. We should see 'greatgrandchild-605' has conditions for the first 9 attempts before
+		// #605 is fixed, and should see 'greatgrandchid-605' has no conditions for all 10 attemps after the fix.
```

Comment:
The original test has a loop here with N=10 and sleep 1 second each round and the e2e test should have a loop of 10 and  make sure there's no condition in each round.

Reply:
Discussed offline that we can actually omit the loop since after the fix the obsolete condition should be gone immediately (before the issue is fixed, it takes about 8s for the condition to be gone).

So we just need to update the comment to mention why we have `sleep 30` before and the reason is that it should trigger that the obsolete condition still stays for 8s if the issue is NOT fixed.

Reply:
done